packages/subapp-server/lib/setup-hapi-routes.js

```diff
@@ -106,17 +106,25 @@ function setupRouteRender({ subAppsByPath, srcDir, routeOptions }) {
   // load subapps for the route
   if (routeOptions.subApps) {
     routeOptions.__internals.subApps = [].concat(routeOptions.subApps).map(x => {
-      let options = {};
+      let options;
       if (Array.isArray(x)) {
         options = x[1];
         x = x[0];
       }
       // absolute: use as path
+      // module: resolve module path
       // else: assume dir under srcDir
-      // TBD: handle it being a module
+      if (!x.startsWith(".") && !x.startsWith("/")) {
+        const subAppPath = optionalRequire.resolve(x);
+        if (subAppPath) {
+          const { manifest, subAppOptions } = require(x);
+          x = manifest ? Path.dirname(subAppPath) : x;
+          options = options || subAppOptions;
```

Comment:
Added this to support passing subapp options from main app also. This will be useful in use-cases where main app wants to customize the options for individual subapps.